benchmark/src/util.ts

```diff
@@ -0,0 +1,71 @@
+import { BenchmarkFunction } from "./benchmark_types";
+
+export function toFixed(num: number, decimalPlaces = 0): string {
+    return string.format(`%.${decimalPlaces}f`, num);
+}
+
+export function calculatePercentageChange(oldValue: number, newValue: number): number {
+    return (newValue / oldValue) * 100 - 100;
+}
+
+// @ts-ignore
+export const isWindows = package.config.sub(1, 1) === "\\";
+
+export const json: {
+    decode: (this: void, str: string) => {};
+    encode: (this: void, val: any) => string;
+} = require("json");
+
+export function readFile(path: string): string {
+    const fileOpenArray = io.open(path, "rb");
+
+    if (fileOpenArray?.[0]) {
+        const fileHandle = fileOpenArray[0];
+        const fileContent = readAll(fileHandle);
+        fileHandle.close();
+
+        return fileContent;
+    }
+    throw Error(`Can't open file ${path}`);
```

Comment:
The optional chaining in `fileOpenArray?.[0]` would just get ignored since we don't support it now. Also, `io.open` appears to return a tuple, which never can be null. Since code flow analysis isn't require there, we can just destructure it in-place
```suggestion
    const [fileHandle] = io.open(path, "rb");
    if (!fileHandle) {
        throw Error(`Can't open file ${path}`);
    }

    const fileContent = readAll(fileHandle);
    fileHandle.close();
    return fileContent;
```

Reply:
The linter forced me :(